### PR TITLE
doc: Add `Safety Enhancements of Unsafe Rust` report

### PR DESCRIPTION
**Pull request by bitboom**
This PR adds the `Safety Enhancements of Unsafe Rust (SEUR)` report, detailing the improvements made to enhance the safety of `unsafe` Rust code within our project.

The report includes an analysis of the [Islet v1.0](https://github.com/islet-project/islet/tree/v1.0) state, the applied safety practices, and the results of the enhancements (refer to [Islet v1.0-seur](https://github.com/islet-project/islet/tree/v1.0-seur)).

Please refer [here](https://github.com/islet-project/islet/blob/seur/doc/rust-unsafe-analysis.md) for details.

### CODE REVIEW

**Review by zpzigi754 — COMMENTED** [review]
Good to include this report! How about changing the name of document and folder from `seur` to `rust-unsafe-analysis` or something else?

**Comment by hihi-wang** [comment]
Nice to have fancy docs for unsafe report!

**Review by hihi-wang — APPROVED** [review]

File: doc/seur.md [hunk 1]
```diff
@@ -0,0 +1,251 @@
+# Safety Enhancements of Unsafe Rust
+
+## Overview <a name="overview"></a>
+
+### **Islet: An On-Device Confidential Computing Platform**
+
+Islet is an open-source project written in Rust that enables confidential computing on ARM architecture devices using the ARMv9 CCA \[0\]. Islet consists of the three main components (refer to Fig. 0). The `RMM` operates at EL2 in the Realm world on the application processor cores and manages the confidential VMs, known as realms. The `HES` performs device boot measurement, generates platform attestation reports, and manages sealing key functionality within a secure hardware IP apart from the main application processor. The `SDK` provides confidential computing primitives to realms.
```
Comment (bokdeuk-jeong) [inline doc/seur.md:7]: suggestion "two main components and a sdk".

**Comment by jinbpark** [comment]
> How about changing the name of document and folder from `seur` to `rust-unsafe-analysis` or something else?

I vote for this as well.

File: doc/seur.md (hunk 1 above)
Comment (bitboom) [reply doc/seur.md]: Reflected. Thanks :)

**Comment by bitboom** [comment]
> Good to include this report! How about changing the name of document and folder from `seur` to `rust-unsafe-analysis` or something else?

I fixed it to `rust-unsafe-analysis`. Thanks!

**Pull request merged**